src/main/java/uk/gov/hmcts/ccd/domain/model/search/elasticsearch/ElasticsearchRequest.java

```diff
@@ -59,22 +93,64 @@ public JsonNode getSource() {
                 .map(JsonNode::asText)
                 .map(this::getFieldId)
                 .filter(Objects::nonNull)
-                .collect(Collectors.toList());
+                .collect(toList());
         }
 
         return Collections.emptyList();
     }
 
     public boolean hasQuery() {
-        return searchRequest.has(QUERY);
+        return nativeSearchRequest.has(QUERY);
     }
 
     public JsonNode getQuery() {
-        return searchRequest.get(QUERY);
+        return nativeSearchRequest.get(QUERY);
+    }
+
+    public boolean hasSupplementaryData() {
+        return supplementaryData != null && supplementaryData.isArray();
+    }
+
+    public JsonNode getSupplementaryData() {
+        return supplementaryData;
+    }
+
+    public void setSupplementaryData(JsonNode node) {
+        if (node.isArray() && arrayContainsOnlyText((ArrayNode) node)) {
+            this.supplementaryData = (ArrayNode) node;
+        } else {
+            throw new BadSearchRequest("Provided supplementary_data must be an array of text fields.");
+        }
```

Comment:
better move this to:
https://github.com/hmcts/ccd-data-store-api/blob/17f3cbe287ac5de485d0c9a875339dcc908bc40d/src/main/java/uk/gov/hmcts/ccd/domain/service/search/elasticsearch/ElasticsearchQueryHelper.java#L31